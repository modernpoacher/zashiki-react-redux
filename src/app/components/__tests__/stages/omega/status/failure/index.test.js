import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Failure from '@modernpoacher/zashiki-react-redux/app/components/stages/omega/status/failure'

Enzyme.configure({ adapter: new Adapter() })

describe('@modernpoacher/zashiki-react-redux/app/components/stages/omega/status/failure', () => {
  describe('Always', () => {
    it('renders', () => {
      const component = (
        <Failure name={'MOCK NAME'} onSubmit={jest.fn()} />
      )

      expect(shallow(component))
        .toMatchSnapshot()
    })
  })

  describe('`code` is an integer', () => {
    it('renders', () => {
      const component = (
        <Failure code={1} name={'MOCK NAME'} onSubmit={jest.fn()} />
      )

      expect(shallow(component))
        .toMatchSnapshot()
    })
  })

  describe('`message` is a string', () => {
    it('renders', () => {
      const component = (
        <Failure name={'MOCK NAME'} message={'MOCK MESSAGE'} onSubmit={jest.fn()} />
      )

      expect(shallow(component))
        .toMatchSnapshot()
    })
  })
})
