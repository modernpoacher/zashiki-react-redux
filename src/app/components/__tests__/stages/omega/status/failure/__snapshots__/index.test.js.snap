// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/omega/status/failure \`code\` is an integer renders 1`] = `ShallowWrapper {}`;

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/omega/status/failure \`message\` is a string renders 1`] = `ShallowWrapper {}`;

exports[`@modernpoacher/zashiki-react-redux/app/components/stages/omega/status/failure Always renders 1`] = `ShallowWrapper {}`;
